src/tools/create-app-tester.js

```diff
@@ -45,11 +45,15 @@ const createAppTester = appRaw => {
 
     const method = resolveMethodPath(appRaw, methodOrFunc);
 
+    // we need a value that's unique to this test (but consistent across test runs) so we don't lose cursors
+    const username = process.env.USER || process.env.USERNAME || 'fallback';
+    const unique = md5(`${method}-${username}`);
+
     const event = {
       command: 'execute',
       method,
       bundle,
-      storeKey: shouldPaginate(appRaw, method) ? `testKey-${genId()}` : null
+      storeKey: shouldPaginate(appRaw, method) ? `testKey-${unique}` : null
```

Comment:
I updated the OP (since comments always get lost). Let me know if it doesn't make sense

Reply:
Really appreciate the detailed explanation! That does make a lot of sense. Since devs are hitting Zapier server when they run tests with `z.cursor`, we need a way to separate cursor storage by devs. But `USER` seems a bit fragile. Is it true that if two devs have the same username on their machines, they may collide with each other? Can we use a more unique thing? I was thinking about MAC address, but there doesn't seem to be a cross-platform way for Node.js to get it. How about we keep using `genId()`? But instead of calling `getId()` on every `appTester(method, bundle)` call, we only call `genId()` once when `createAppTester` is called. Here's the idea:

```js
const createAppTester = appRaw => {
  const handler = createLambdaHandler(appRaw);
  const createHandlerPromise = promisifyHandler(handler);

  // A unique ID for this test run
  const testId = getId();  // probably better to use randomString() to be more unique

  return (methodOrFunc, bundle) => {
    bundle = bundle || {};

    const method = resolveMethodPath(appRaw, methodOrFunc);

    // we need a value that's unique to this test (but consistent across test runs) so we don't lose cursors
    const unique = md5(`${method}-${testId}`);

    const event = {
      command: 'execute',
      method,
      bundle,
      storeKey: shouldPaginate(appRaw, method) ? `testKey-${unique}` : null
    };

    // ...
  };
};

```

Reply:
Ah, that'll do it! I worked myself into a loop at made it a lot more complex than it needed to be. 

Also FWIW, collision risk here would have only been for users with identical usernames and test methods who run their cursor tests within 60 seconds of each other. not zero, but low enough. 

In any case, I think your solution is better! I've pushed that change.